lesson16/main.js

```diff
@@ -0,0 +1,125 @@
+   const json = await response.json();
+   return json.data;
+}
+
+const createElementWithClassName = (element, name) => {
+   const createElement = document.createElement(element);
+   createElement.classList.add(name);
+   return createElement;
+};
+
+const displayMessage = (message) => {
+   const div = document.createElement("div");
+   body.appendChild(div);
+   div.textContent = message;
+}
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   body.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+createTabContents();
+async function createTabContents() {
+   startLoading();
+   let responseData;
+   try {
+      responseData = await fetchData(jsonURL);
+      console.log(responseData);
+      if(responseData.length === 0) {
+         displayMessage("データはまだありません");
+         return;
+         } 
+   } catch (error) {
+      displayMessage(error);
+      console.log(error);
+   } finally {
+      endLoading();
+   }
+   createCategoryTab(responseData);
+   renderTabContainer(responseData);
+}
+
+const createCategoryTab = (values) => {
+   const fragment = document.createDocumentFragment();
+   for (let i = 0; i < values.length; i++) {
+      const categoryTabItem = createElementWithClassName("li", "category-tab__item")
+      categoryTabItem.id = `js-categoryTabItem${i}`;
+      const categoryTabText = createElementWithClassName("a", "category-tab__text");
+      categoryTabText.id = `js-categoryTabText${i}`
+      categoryTabText.textContent = values[i].category
+      fragment.appendChild(categoryTabItem).appendChild(categoryTabText);
+   }
+   categoryTab.appendChild(fragment);
+   //初期表示のタブの選択はデータとして持っているがまだ理解できていないので、現段階ではNewsのタブとしてある
+   const newsTabItem = document.getElementById("js-categoryTabItem0");
+   newsTabItem.classList.add("is-active");
+}
+
+const renderTabContainer = (values) => {
+   const tabFragment = document.createDocumentFragment();
+   const tabContainer = createElementWithClassName("div", "tab-container");
+   tabContainer.id = "js-tabContainer";
+   const articlesData = values.map((value) => value.articles);
+   for (let i = 0; i < values.length; i++) {
+      const tabContents = createElementWithClassName("div", "tab-contents");
+      tabContents.id = `js-tabContents${i}`;
+      const articlesTitles = articlesData[i].map((value) => value.title);
+      const articlesHref = articlesData[i].map((value) => value.href);
+      const tabLists = createElementWithClassName("ul", "tab-contents__lists");
+      tabLists.id = `js-tabLists${i}`;
+      const tabImage = createElementWithClassName("img", "tab-contents__image");
+      tabImage.id = `js-tabImage${i}`;
+      tabImage.src = values[i].img;
+      console.log(values);
+      console.log(articlesTitles);
```

Comment:
MUST
Please delete the unnecessary console.log

Reply:
Thank you for pointing that out. 
I removed that.
[ec8b8f8](https://github.com/taka0209/js_lesson/pull/18/commits/ec8b8f85bc532e47b37f12077105ac7e58e60ab8)